### ISSUE TITLE
NitfiMasker Not removing confounds 

### ISSUE THREAD

**Issue by DasDominus** [nilearn/nilearn#3434]
So I tried to extract signal from the fMRI image with a given mask
and loading confounds retrieved from fmriprep. below is what I did:

The issue is, extracted signal when cross-correlated, shows significant impact from low-drift signals
![image](https://user-images.githubusercontent.com/76655953/204946116-8354cf77-187c-493b-8f79-c517eb8d4a05.png)

This is very likely the result from not having done high_pass filtering.
However, I have passed in high_pass when loading confounds and I can confirm these are populated in fmriprep results.

Nilearn version:

### Expected behavior
Get signals that are high_pass filtered
![image](https://user-images.githubusercontent.com/76655953/204946374-7926ec74-5072-4b92-b06e-2e172243c1db.png)

### Actual behavior
data are likely not high_pass filtered

### Steps and code to reproduce bug

```python
        data_confounds_simple, sample_mask = load_confounds(
        data,
        strategy=["compcor", "high_pass", "motion", "wm_csf"],
        motion="basic", wm_csf="basic", global_signal="basic")

                mask_img = resample_to_img(
                  path,
                  data[f'{conditions[0]}_mask'],
                  interpolation='nearest'
                )

              masker = NiftiMasker(
                  mask_img=mask_img,
                  standardize=False,
                  memory=f"{path}/nilearn_cache")

            raw_signal = masker.fit_transform(
                        imgs, confounds=confounds
                    )
            pd.DataFrame(raw_signal).to_csv(fn)

```

**Comment by DasDominus** [comment]
yeah confirmed this by passing explicit high_pass filter (1/128) to signal.clean

Not sure why high_pass in confounds didn't do the cleaning

**Comment by htwangtw** [comment]
Putting some relevant part of the code for discussion. 
If `NiftiMasker` is having problem with filtering, all maskers would have the same issue, as they all use this base function to pass extracted signal to `signal.clearn`:
https://github.com/nilearn/nilearn/blob/167052cc5457c53a2bd068df75e4f8151f0d7e33/nilearn/maskers/base_masker.py#L25

For high pass filtering, fMRIPrep provides cosine drift terms and nilearn function simply regressing out the regressors. This is a different high-pass filtering method comparing to the butterworth filter accessed through the `high_pass` parameter in `signal.cleans`

In other words, this can be a fMRIPrep issue, but I will have to have a further look. 

I would be curious to see if we generate cosine drift terms, instead of using fMRIPrep supplied ones, what difference will it make.

**Comment by htwangtw** [comment]
> yeah confirmed this by passing explicit high_pass filter (1/128) to signal.clean
> 
> Not sure why high_pass in confounds didn't do the cleaning

@DasDominus Which method? The default is butterworth and there is an option for cosine drift terms. If you used butterworth, can you report back what is it like for the cosine option?

**Comment by DasDominus** [comment]
thanks! @htwangtw 

yeah, it seems it's a different approach.
Just tried with
```python

rr1 = nilearn.signal.clean(rr1.to_numpy(),  high_pass=1/128, filter='cosine')
```
![image](https://user-images.githubusercontent.com/76655953/204949912-eb00ef6e-9b4f-43c7-ba59-cfd962b063f1.png)

but also this could be due to I'm passing in high_pass threshold.

However, I don't think a filter option is provided for nilearn.maskers.NiftiMasker so seems it's using default aka butterworth

**Comment by DasDominus** [comment]
I lack the expertise to debug this, tho I'm happy to provide the data & fmriprep confounds used for reproduction
let me know if that's needed

**Comment by htwangtw** [comment]
I got some code to show the cosine option and fmriprep cosine term creates similar results:

```python
from nilearn.datasets import fetch_development_fmri, fetch_atlas_basc_multiscale_2015
from nilearn.interfaces.fmriprep import load_confounds
from nilearn.maskers import NiftiLabelsMasker
from nilearn.signal import clean
import matplotlib.pyplot as plt


data = fetch_development_fmri(n_subjects=1, reduce_confounds=False)
atlas = fetch_atlas_basc_multiscale_2015(version='asym')

atlas_path = atlas.scale064
img = data.func[0]
high_pass_fmriprep, _ = load_confounds(img, strategy=['high_pass'])

masker = NiftiLabelsMasker(atlas_path)
raw_signal = masker.fit_transform(img)

fmriprep_cleaned = clean(raw_signal, confounds=high_pass_fmriprep)
nilearn_cleaned = clean(raw_signal, filter='cosine', high_pass=1/128)

plt.plot(fmriprep_cleaned[:, 0])
plt.plot(nilearn_cleaned[:, 0])
```
![image](https://user-images.githubusercontent.com/13743617/204956702-850cbabf-a553-4197-9daf-b97db33b5596.png)

**Comment by htwangtw** [comment]
To conclude, I cannot reproduce the problem described by @DasDominus. 
This question might be more appropriate to be discussed on https://neurostars.org/
There's a lot to unpack before we can conclude this is an issue in nilearn, or fmriprep implementation.

**Comment by DasDominus** [comment]
thanks @htwangtw ! I suspect behavior might differ when directly using NiftiMasker to get raw signals
I'll try to repo this weekend
